core/arch/arm/kernel/boot.c

```diff
@@ -1069,8 +1069,15 @@ void __weak boot_init_primary_late(unsigned long fdt __unused,
 
 	boot_primary_init_intc();
 	init_vfp_nsec();
-	if (!IS_ENABLED(CFG_NS_VIRTUALIZATION))
+	if (!IS_ENABLED(CFG_NS_VIRTUALIZATION)) {
+		if (IS_ENABLED(CFG_UNMASK_NATIVE_INTR_AT_BOOT)) {
+			/* Unmask native interrupts during driver initcalls */
+			thread_set_exceptions(thread_get_exceptions() &
+					      ~THREAD_EXCP_NATIVE_INTR);
+		}
+
 		init_tee_runtime();
+	}
```

Comment:
We should probably disable the interrupts again since `entry_a64.S` doesn't expect interrupts to be enabled when `boot_init_primary_late()` returns.

Reply:
Right. Then i'll unmask them back in `boot_init_primary_final()` for `call_driver_initcalls()` and `call_finalcalls()` execution.